src/libcrun/utils.c

```diff
@@ -1102,7 +1102,11 @@ open_unix_domain_client_socket (const char *path, int dgram, libcrun_error_t *er
       path = name_buf;
     }
 
-  strcpy (addr.sun_path, path);
+  if (strlen (path) >= sizeof (addr.sun_path))
```

Comment:
since you are already computing the strlen().  Could we just store it in a variable and then use memcpy instead?  That would be faster (and we don't have to drop the check for strncpy in `make syntax-check`)